### PR TITLE
[DO NOT MERGE YET] Feature/new interface

### PR DESCRIPTION
**Pull request by csavage-mdsol**
This PR is made against the `brave_new_develop` branch, and is open early for comments, discussion and objections.  It implements integration tests against my local copy of `crichton_test_service` until the outstanding PRs are merged.  It also implements the first bits of the proposed new interface.  It has removed the need for active support so far, and I'd like to keep it that way.  However, this will end up being a significant rewrite, so expect more red to come in the diff.

Specs pass:

```
Finished in 12.12 seconds
42 examples, 0 failures, 1 pending

Randomized with seed 4080
```

### CODE REVIEW

File: lib/farscape/agent.rb [hunk 1]
```diff
@@ -1,60 +1,35 @@
...
+
+    def initialize(entry = nil, media = :hale)
+      @entry_point = entry
+      @media_type = media
     end
 
-    private
-    def build_request(request)
-      Request.build_or_return(request)
+    def enter(entry = nil)
```
Comment (HonoreDB) [inline lib/farscape/agent.rb]: `entry = entry_point`

File: lib/farscape/agent.rb [hunk 2]
```diff
@@ -1,60 +1,35 @@
...
+      raise "No Entry Point Provided!" unless url
+
+      response = client.invoke(url: url, headers: get_accept_header(media_type))
+      Representor.new(media_type, response.body, self)
     end
 
-    class UnregisteredClientError < StandardError; end
+    # TODO share this information with serialization factory base
+    def get_accept_header(media_type)
+      media_types = { hale: 'application/vnd.hale+json' }
```
Comment (HonoreDB) [inline lib/farscape/agent.rb:25]: Hashes like this should probably live as constants somewhere.

Comment (csavage-mdsol) [inline lib/farscape/agent.rb:25]: It does in Representors, we haven't yet decided where to pull it in.  Just left the TODO above for now.

File: lib/farscape/agent.rb [hunk 3]
```diff
@@ -1,60 +1,33 @@
...
-      Request.build_or_return(request)
+    # TODO share this information with serialization factory base
+    def get_accept_header(media_type)
+      media_types = { hale: 'application/vnd.hale+json' }
+      { 'Accept' => media_types[media_type] }
     end
 
-    class UnregisteredClientError < StandardError; end
+    def client
+      { http: HTTPClient }[PROTOCOL].new
```
Comment (HonoreDB) [inline lib/farscape/agent.rb:30]: Can this be memoized?

File: farscape.gemspec [hunk 4]
```diff
@@ -17,9 +17,10 @@ Gem::Specification.new do |s|
     Farscape is a library that simplifies consuming Hypermedia API responses.
   DESC
 
-  s.add_dependency('rake')
-  s.add_dependency('activesupport', '>= 3.2.0')
+  s.add_dependency 'dice_bag'
```
Comment (jcarres-mdsol) [inline farscape.gemspec:20]: should add version

**Comment by csavage-mdsol** [comment]
Closing in preference of the newly winnowed nearscape branch.

**Pull request closed without merging**